openmc/data/multipole.py

```diff
@@ -0,0 +1,628 @@
+    def w_start(self):
+        return self._w_start
+
+    @property
+    def w_end(self):
+        return self._w_end
+
+    @property
+    def broaden_poly(self):
+        return self._broaden_poly
+
+    @property
+    def curvefit(self):
+        return self._curvefit
+
+    @num_l.setter
+    def num_l(self, num_l):
+        if num_l is not None:
+            cv.check_type('num_l', num_l, Integral)
+            cv.check_greater_than('num_l', num_l, 1, equality=True)
+            cv.check_less_than('num_l', num_l, 4, equality=True)
+            # There is an if block in _evaluate that assumes num_l <= 4.
+        self._num_l = num_l
+
+    @fit_order.setter
+    def fit_order(self, fit_order):
+        if fit_order is not None:
+            cv.check_type('fit_order', fit_order, Integral)
+            cv.check_greater_than('fit_order', fit_order, 2, equality=True)
+            # _broaden_wmp_polynomials assumes the curve fit has at least 3
+            # terms.
+        self._fit_order = fit_order
+
+    @fissionable.setter
+    def fissionable(self, fissionable):
+        if fissionable is not None:
+            cv.check_type('fissionable', fissionable, bool)
+        self._fissionable = fissionable
+
+    @formalism.setter
+    def formalism(self, formalism):
+        if formalism is not None:
+            cv.check_type('formalism', formalism, string_types)
+            cv.check_value('formalism', formalism, ('MLBW', 'RM'))
+        self._formalism = formalism
+
+    @spacing.setter
+    def spacing(self, spacing):
+        if spacing is not None:
+            cv.check_type('spacing', spacing, Real)
+            cv.check_greater_than('spacing', spacing, 0.0, equality=False)
+        self._spacing = spacing
+
+    @sqrtAWR.setter
+    def sqrtAWR(self, sqrtAWR):
+        if sqrtAWR is not None:
+            cv.check_type('sqrtAWR', sqrtAWR, Real)
+            cv.check_greater_than('sqrtAWR', sqrtAWR, 0.0, equality=False)
+        self._sqrtAWR = sqrtAWR
+
+    @start_E.setter
+    def start_E(self, start_E):
+        if start_E is not None:
+            cv.check_type('start_E', start_E, Real)
+            cv.check_greater_than('start_E', start_E, 0.0, equality=True)
+        self._start_E = start_E
+
+    @end_E.setter
+    def end_E(self, end_E):
+        if end_E is not None:
+            cv.check_type('end_E', end_E, Real)
+            cv.check_greater_than('end_E', end_E, 0.0, equality=False)
+        self._end_E = end_E
+
+    @data.setter
+    def data(self, data):
+        if data is not None:
+            cv.check_type('data', data, np.ndarray)
+            if len(data.shape) != 2:
+                raise ValueError('Multipole data arrays must be 2D')
+            if data.shape[1] not in (4, 5):  # 4 for RM, 5 for MLBW
+                raise ValueError('The second dimension of multipole data arrays'
+                                 ' must have a length of either 4 or 5')
+            if not np.issubdtype(data.dtype, np.complex):
```

Comment:
`np.complex` is simply the same as `complex`. Also, we do have some logic in `cv.check_type` to handle ndarrays, so you could use that.